spec/support/http_handling_shared.rb

```diff
@@ -148,6 +148,18 @@
         expect(sockets_used.uniq.length).to eq(1)
       end
 
+      context "on a mixed state" do
+        it "re-opens the connection" do
+          first_socket = client.get("#{server.endpoint}/socket/1").body.to_s
```

Comment:
Sorry for that useless noise. Had not enough of sleep. Will try to avoid commenting while in a grumpy old man mode next time. :D Or at least will try to get into details prior making any judgements. :D Sorry once again.